### ISSUE TITLE
Wrong Indentation for the `Layout/MultilineMethodCallIndentation` cop

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12428]
## Actual behavior

```ruby
def parse_user_params
  params[:user] =
    params[:user]&.permit(*policy(User).permitted_attributes).to_h
    .merge(
      params[:user_form]
      &.permit(*policy(UserForm).permitted_attributes)
      .to_h,
    )
end
```

I run the command `rubocop -A` and I get:

```ruby
def parse_user_params
  params[:user] =
    params[:user]&.permit(*policy(User).permitted_attributes).to_h
                 .merge(
                   params[:user_form]
                   &.permit(*policy(UserForm).permitted_attributes)
                   .to_h,
                 )
end
```

## Expected behavior

```ruby
def parse_user_params
  params[:user] =
    params[:user]
    &.permit(*policy(User).permitted_attributes)
    .to_h
    .merge(
      params[:user_form]
      &.permit(*policy(UserForm).permitted_attributes)
      .to_h,
    )
end
```
or
```ruby
def parse_user_params
  params[:user] =
    params[:user]&.permit(*policy(User).permitted_attributes)
                 .to_h
                 .merge(
                   params[:user_form]
                   &.permit(*policy(UserForm).permitted_attributes)
                   .to_h,
                 )
end
```

## RuboCop version

```console
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop -V
1.58.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 2.7.8) [x86_64-linux]
  - rubocop-capybara 2.19.0
  - rubocop-factory_bot 2.24.0
  - rubocop-performance 1.19.1
  - rubocop-rails 2.22.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.25.0
  - rubocop-thread_safety 0.5.1
```

**Comment by ydakuka** [comment]
And the second case:

I have the following code:

```ruby
def parse_user_params
  params[:user] =
    params[:user]&.permit(*policy(User).permitted_attributes)&.to_h
    &.merge(
      params[:user_form]
      &.permit(*policy(UserForm).permitted_attributes)
      .to_h,
    )
end
```

I run `rubocop -A` and get:

```ruby
def parse_user_params
  params[:user] =
    params[:user]&.permit(*policy(User).permitted_attributes)&.to_h
                 &.merge(
                   params[:user_form]
    &.permit(*policy(UserForm).permitted_attributes)
    .to_h,
                 )
end
```

**Comment by koic** [comment]
`Layout/MultilineMethodCallIndentation` is for organizing the indentation of multiline method calls that are already broken into multiple lines, not for introducing new line breaks in method chains.
